kernel/include/kswap.h

```diff
@@ -126,7 +126,7 @@ static inline int z_swap_irqlock(unsigned int key)
 #endif
 #endif
 	ret = __swap(key);
-#ifndef CONFIG_ARM
+#if !defined(CONFIG_ARM) || !defined(CONFIG_ARC)
```

Comment:
`#if !defined(CONFIG_ARM) && !defined(CONFIG_ARC)`
Otherwise his code will be executed always unless both symbols are defined.

Reply:
Fixed